### PR TITLE
Ran clang-tidy with modernize-use-using

### PR DESCRIPTION
**Pull request by bryancall**

### CODE REVIEW

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/304/

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/1892/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/317/

**Comment by atsci** [comment]
FreeBSD11 build *failed*! https://ci.trafficserver.apache.org/job/freebsd-github/1999/

**Comment by atsci** [comment]
Intel CC build *failed*! https://ci.trafficserver.apache.org/job/icc-github/430/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/301/

**Comment by atsci** [comment]
clang-analyzer build *successful*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/563/

**Comment by atsci** [comment]
clang format *failed*! https://ci.trafficserver.apache.org/job/clang-format-github/305/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/318/

**Comment by atsci** [comment]
FreeBSD11 build *failed*! https://ci.trafficserver.apache.org/job/freebsd-github/2000/

**Comment by atsci** [comment]
Intel CC build *failed*! https://ci.trafficserver.apache.org/job/icc-github/431/

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/1893/

**Comment by atsci** [comment]
AU check *failed*! https://ci.trafficserver.apache.org/job/autest-github/302/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/564/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/307/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/320/

**Comment by atsci** [comment]
clang format *successful*! https://ci.trafficserver.apache.org/job/clang-format-github/308/

**Comment by atsci** [comment]
Intel CC build *failed*! https://ci.trafficserver.apache.org/job/icc-github/433/

**Comment by atsci** [comment]
Linux build *failed*! https://ci.trafficserver.apache.org/job/linux-github/1895/

**Comment by atsci** [comment]
FreeBSD11 build *failed*! https://ci.trafficserver.apache.org/job/freebsd-github/2002/

**Comment by atsci** [comment]
RAT check *successful*! https://ci.trafficserver.apache.org/job/RAT-github/321/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/304/

**Comment by atsci** [comment]
FreeBSD11 build *successful*! https://ci.trafficserver.apache.org/job/freebsd-github/2003/

**Comment by atsci** [comment]
Intel CC build *successful*! https://ci.trafficserver.apache.org/job/icc-github/434/

**Comment by atsci** [comment]
AU check *successful*! https://ci.trafficserver.apache.org/job/autest-github/305/

**Comment by atsci** [comment]
Linux build *successful*! https://ci.trafficserver.apache.org/job/linux-github/1897/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/566/

**Comment by atsci** [comment]
clang-analyzer build *failed*! https://ci.trafficserver.apache.org/job/clang-analyzer-github/567/

**Comment by bryancall** [comment]
using is a superset of typedef and can be used for templates.

**Comment by bryancall** [comment]
[approve ci]

File: proxy/http2/test_Http2DependencyTree.cc [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
 using namespace std;
 
-typedef Http2DependencyTree<string *> Tree;
+using Tree = Http2DependencyTree<std::__cxx11::string *>;
```
Comment (jpeach) [inline proxy/http2/test_Http2DependencyTree.cc]: This one is wrong.

File: example/ssl-preaccept/ssl-preaccept.cc [hunk 2]
```diff
@@ -43,7 +43,7 @@ namespace
 {
 std::string ConfigPath;
 typedef std::pair<IpAddr, IpAddr> IpRange;
-typedef std::deque<IpRange> IpRangeQueue;
+using IpRangeQueue = std::deque<IpRange>;
```
Comment (jpeach) [inline example/ssl-preaccept/ssl-preaccept.cc]: Why not do both typedefs?

Comment (bryancall) [inline example/ssl-preaccept/ssl-preaccept.cc]: clang-tidy isn't 100%.  There will need to be a followup commit to fix the remaining items.

**Pull request merged**